internal/services/network/network_manager_management_group_connection_resource.go

```diff
@@ -0,0 +1,245 @@
+	return map[string]*pluginsdk.Schema{
+		"connection_state": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+	}
+}
+
+func (r ManagerManagementGroupConnectionResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model ManagerManagementGroupConnectionModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			client := metadata.Client.Network.ManagerManagementGrpConnectionsClient
+			managementGroupId, err := managementParse.ManagementGroupID(model.ManagementGroupId)
+			if err != nil {
+				return err
+			}
+
+			id := parse.NewNetworkManagerManagementGroupConnectionID(managementGroupId.Name, model.Name)
+			existing, err := client.Get(ctx, id.ManagementGroupName, id.NetworkManagerConnectionName)
+			if err != nil && !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("checking for existing %s: %+v", id, err)
+			}
+
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			managerConnection := &network.ManagerConnection{
+				ManagerConnectionProperties: &network.ManagerConnectionProperties{},
+			}
+
+			if model.Description != "" {
+				managerConnection.ManagerConnectionProperties.Description = &model.Description
+			}
+
+			if model.NetworkManagerId != "" {
+				managerConnection.ManagerConnectionProperties.NetworkManagerID = &model.NetworkManagerId
+			}
+
+			if _, err := client.CreateOrUpdate(ctx, *managerConnection, id.ManagementGroupName, id.NetworkManagerConnectionName); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r ManagerManagementGroupConnectionResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Network.ManagerManagementGrpConnectionsClient
+
+			id, err := parse.NetworkManagerManagementGroupConnectionID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var model ManagerManagementGroupConnectionModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			existing, err := client.Get(ctx, id.ManagementGroupName, id.NetworkManagerConnectionName)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			properties := existing.ManagerConnectionProperties
+			if properties == nil {
+				return fmt.Errorf("retrieving %s: properties was nil", id)
+			}
+
+			if metadata.ResourceData.HasChange("description") {
+				if model.Description != "" {
+					properties.Description = &model.Description
+				} else {
+					properties.Description = nil
+				}
```

Comment:
why do we need to explicitly nil this? these fields are `omitempty` so this won't be sent..